vizro-core/src/vizro/actions/export_data_class_action.py

```diff
@@ -0,0 +1,110 @@
+import importlib
+from typing import Any, Dict, List, Literal
+
+from dash import Output, State, ctx, dcc
+
+from vizro.actions._filter_action import _filter
+from vizro.actions.filter_interaction_action import filter_interaction
+from vizro.managers import model_manager
+from vizro.models.types import CapturedActionCallable
+
+
+class ExportDataClassAction(CapturedActionCallable):
+    def __init__(self, *args, **kwargs):
+        self._args = args
+        self._kwargs = kwargs
+        # Fake initialization - to let other actions see that this one exists.
+        super().__init__(*args, **kwargs)
+
+    def _post_init(self):
```

Comment:
Should this be called directly in `__init__` or is that too early?

Reply:
It's too early to call `_post_init` validation inside the action initialisation phase. To be able to validate action input arguments properly, all dashboard models have to be initialised. Some of the models are initialised in the `_pre_build` phase of other models which means that the actions `_post_init` has to be called within the build phase.